js/dialogcards.js

```diff
@@ -70,26 +92,137 @@ H5P.Dialogcards = (function ($, Audio, JoubelUI) {
+        existsCardOrder = false;
+      }
+      // Create cardOrder and cardNumber buttons only on first instanciation for logged in user.
+      if (self.params.behaviour.randomCards == 'user' && !existsCardOrder) {
+        self.createOrder().appendTo(self.$inner);
+      } else {
+        self.attachContinue();
+      }
+  };
+
+    /**
+   * Attach the rest of the h5p inside the given container.
+   *
+   * @param {jQuery} $container
+   */
+   C.prototype.attachContinue = function ($container) {
+    var self = this;
+
+    // Remove potential user interaction elements from DOM.
+    $( '.h5p-dialogcards-number', self.$inner ).remove();
+
     if (self.params.behaviour.scaleTextNotCard) {
-      $container.addClass('h5p-text-scaling');
+      self.$inner.addClass('h5p-text-scaling');
     }
 
-    self.initCards(self.params.dialogs)
+    self.initCards(self.dialogs)
       .appendTo(self.$inner);
 
     self.createFooter()
       .appendTo(self.$inner);
 
     self.updateNavigation();
 
-    self.on('reset', function () {
-      self.reset();
+    self.on('retry', function () {
+      self.retry();
+    });
+
+    self.on('resetTask', function () {
+      self.resetTask();
     });
 
     self.on('resize', self.resize);
     self.trigger('resize');
   };
 
+
+  /**
+   * Create orderCards option request
+   *
+   * @returns {*|jQuery|HTMLElement} Order element
+   */
+  C.prototype.createOrder = function () {
+    var self = this;
+    var $order = $('<div>', {
+      'class': 'h5p-dialogcards-order',
+      'style': 'height:200px;',
+      'html': self.params.randomizeCardsQuestion + "&nbsp;&nbsp;",
+      'style': 'height:200px;'
+    });
+
+    self.$normalOrder = JoubelUI.createButton({
+      'class': 'h5p-dialogcards-order-button',
+      'title': self.params.no,
+      'html': self.params.no
+    }).click(function () {
+      self.randomizeOrder("normal");
+    }).appendTo($order);
+
+    self.$randomizeOrder = JoubelUI.createButton({
+      'class': 'h5p-dialogcards-order-button',
+      'title': self.params.yes,
+      'html': self.params.yes
+    }).click(function () {
+      self.randomizeOrder("random");
+    }).appendTo($order);
+
+    return $order;
+  };
+
+
+  /**
+   * Create numberCards option request
+   *
+   * @returns {*|jQuery|HTMLElement} numberCards element
+   */
+  C.prototype.createNumberCards = function () {
+    var self = this;
+
+    var numCards = self.params.dialogs.length;
+
+    var $numberCards = $('<div>', {
+      'class': 'h5p-dialogcards-number',
+      'style': 'height:200px;',
+      'html': self.params.nbCardsQuestion + "<br /><br />"
+    });
+
+    /**
+     * Allow user to select a number of cards to play with, by displaying selectable buttons in increments of 5.
+    **/
+    for (var i = 5; i < numCards; i+= 5) {
+      self.$Button = JoubelUI.createButton({
+          'class': 'h5p-dialogcards-number-button',
+          'title': i,
+          'html': i,
+          'id': i
```

Comment:
I have tested several DC instances on the same page, in "User Choice" mode and have found no conflict with multiple identical ids for those cards number buttons. And I do not know how I could do otherwise.